SU2_CFD/src/output/CMultizoneOutput.cpp

```diff
@@ -87,55 +87,39 @@ CMultizoneOutput::~CMultizoneOutput() {}
 
 void CMultizoneOutput::LoadMultizoneHistoryData(COutput **output, CConfig **config) {
 
-  unsigned short iZone, iField, nField;
+  unsigned short iZone;
   string name, header;
 
   for (iZone = 0; iZone < nZone; iZone++){
 
-    map<string, HistoryOutputField> ZoneHistoryFields = output[iZone]->GetHistoryFields();
-    vector<string>                  ZoneHistoryNames  = output[iZone]->GetHistoryOutput_List();
-
-    nField = ZoneHistoryNames.size();
-
-
-    /*-- For all the variables per solver --*/
-    for (iField = 0; iField < nField; iField++){
-
-      if (ZoneHistoryNames[iField] != "TIME_ITER" && ZoneHistoryNames[iField] != "OUTER_ITER"){
-
-        name   = ZoneHistoryNames[iField]+ "[" + PrintingToolbox::to_string(iZone) + "]";
-
-        SetHistoryOutputValue(name, ZoneHistoryFields[ZoneHistoryNames[iField]].value);
+    const COutFieldCollection& ZoneHistoryFields = output[iZone]->GetHistoryFieldsAll();
 
+    for (auto field : ZoneHistoryFields.GetReferencesAll()){
```

Comment:
I think auto type deduction looses "referenceness" i.e. you may want `auto& field` or `const auto& field` depending on the return type of GetReferencesAll